js/src/main/scala/scalajson/ast/JValue.scala

```diff
@@ -77,22 +77,26 @@ object JNumber {
   def apply(value: Float): JValue = value match {
     case n if java.lang.Float.isNaN(n) => JNull
     case n if n.isInfinity => JNull
-    case _ => JNumber(value.toString)
+    case _ => new JNumber(value.toString)
   }
+
+  def fromString(value: String): Option[JNumber] =
+    if (value.matches(jNumberRegex))
+      Some(new JNumber(value))
+    else
+      None
+
+  def unapply(arg: JNumber): Option[String] = Some(arg.underlying)
```

Comment:
From an end-user standpoint I'm not a fan of having a `JNumber.unapply` that provides a string. Maybe we should just remove the unapply altogether? What is a user supposed to do when getting this string? Try to parse it as a double? as a `BigDecimal`? It would be nice if the type system could help out more here.